e2e/test/PnpServiceTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    [TestCategory("PlugAndPlay")]
+    public class PnpServiceTests
+    {
+        private const string DevicePrefix = "plugAndPlayDevice";
```

Comment:
We prefix our E2E test devices with `E2E_`. 
This makes it easier for the cleanup script to delete these devices periodically: https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/service/CleanUpDevicesSample/csharp_devices_list.csv

Reply:
On the other hand, I'm inclined to add delete device. Does remove device just remove the registration or delete it as well?

Reply:
Ah, you want to handle the create - test - delete within the test directly? I don't see any issue with this, other than if this adds considerable test run duration because of the  addition of the "deletion" operation.

>Does remove device just remove the registration or delete it as well?

I am assuming you mean removing the device registration from IoT hub identity registry. If yes, then "removing the device registration from IoT hub identity registry" = "delete the device from IoT Hub".

Reply:
However, it is possible the test fails so I will add the prefix to the helper class so that we always create a device with the right prefix.

Reply:
I've changed wherever it was straightforward. We can keep fixing anything that looks ugly when we run into it. As it does not affect our tests in any way, should be fine for now.